hyperspy/gui/peakfinder2D.py

```diff
@@ -0,0 +1,288 @@
+import inspect
+from collections import OrderedDict
+import numpy as np
+import matplotlib.pyplot as plt
+from IPython.display import display
+from hyperspy._signals.signal2d import find_peaks_max, find_peaks_minmax, \
+    find_peaks_zaefferer, find_peaks_stat, find_peaks_dog, find_peaks_log
+
+METHODS = [find_peaks_max, find_peaks_minmax, find_peaks_zaefferer,
+           find_peaks_stat, find_peaks_dog, find_peaks_log]
+
+
+class PeakFinderUIBase:
+
+    def __init__(self):
+        self.signal = None
+        self.indices = None
+        self.methods = METHODS
+        self.method_names = [method.__name__ for method in METHODS]
+        self.params = {method.__name__: OrderedDict(
+            [(p.name, p.default) for p in
+             inspect.signature(method).parameters.values() if
+             p.default is not inspect._empty]) for method in METHODS}
+        self._method = self.method_names[0]
+
+    def interactive(self, signal):
+        self.signal = signal
+        self.indices = self.signal.axes_manager.indices
+        self.init_ui()
+
+    def init_ui(self):
+        raise NotImplementedError
+
+    @property
+    def current_method(self):
+        return dict(zip(self.method_names, self.methods))[self._method]
+
+    def get_data(self):
+        return self.signal.inav[self.indices].data
+
+    def get_peaks(self):
+        peaks = self.current_method(self.get_data(),
+                                    **self.params[self._method])
+        return peaks
+
+
+class PeakFinderUIIPYW(PeakFinderUIBase):
+    """
+    Find peaks using a Jupyter notebook-based user interface
+    """
+
+    def __init__(self):
+        super(PeakFinderUIIPYW, self).__init__()
+        self.ax = None
+        self.image = None
+        self.pts = None
+        self.param_container = None
+
+    def init_ui(self):
+        self.create_choices_widget()
+        self.create_navigator()
+        self.create_param_widgets()
+        self.plot()
+
+    def create_choices_widget(self):
+        from ipywidgets import Dropdown
+        dropdown = Dropdown(
+            options=list(self.method_names),
+            value=self._method,
+            description="Method",
+        )
+
+        def on_method_change(change):
+            self._method = dropdown.value
```

Comment:
If I remember correctly here you want `change.new` or something along those lines, right? Currently you don't actually read/use `change`. You could also perform a sanity check here (i.e. validate the new value)